spring-integration-core/src/main/java/org/springframework/integration/handler/management/SimpleMessageHandlerMetrics.java

```diff
@@ -54,83 +47,44 @@
 
 	private volatile String name;
 
-	private volatile String source;
-
-
-	public SimpleMessageHandlerMetrics(MessageHandler handler) {
-		this.handler = handler;
-	}
-
+	private volatile boolean fullStatsEnabled;
 
 	public void setName(String name) {
 		this.name = name;
 	}
 
-	@Override
-	public String getName() {
-		return this.name;
+	public void setFullStatsEnabled(boolean fullStatsEnabled) {
+		this.fullStatsEnabled = fullStatsEnabled;
 	}
 
-	public void setSource(String source) {
-		this.source = source;
-	}
-
-	@Override
-	public String getSource() {
-		return this.source;
-	}
-
-	public MessageHandler getMessageHandler() {
-		return this.handler;
-	}
-
-	@Override
-	public Object invoke(MethodInvocation invocation) throws Throwable {
-		String method = invocation.getMethod().getName();
-		if ("handleMessage".equals(method)) {
-			Message<?> message = (Message<?>) invocation.getArguments()[0];
-			handleMessage(invocation, message);
-			return null;
-		}
-		return invocation.proceed();
-	}
-
-	private void handleMessage(MethodInvocation invocation, Message<?> message) throws Throwable {
+	public long beforeHandle(Message<?> message) throws Exception {
 		if (logger.isTraceEnabled()) {
-			logger.trace("messageHandler(" + this.handler + ") message(" + message + ") :");
+			logger.trace("messageHandler(" + this.name + ") message(" + message + ") :");
 		}
-		String name = this.name;
-		if (name == null) {
-			name = this.handler.toString();
+		long start = 0;
+		if (this.fullStatsEnabled) {
+			start = System.currentTimeMillis();
 		}
-		StopWatch timer = new StopWatch(name + ".handle:execution");
-		try {
-			timer.start();
-			this.handleCount.incrementAndGet();
-			this.activeCount.incrementAndGet();
-
-			invocation.proceed();
+		this.handleCount.incrementAndGet();
+		this.activeCount.incrementAndGet();
```

Comment:
Can you explain the difference between `handleCount` and `activeCount`.
I see that `activeCount` isn't resetted in the `reset()`.
But can't we adjust `activeCount` only on the `reset()` and add with the `handleCount` on the `getActiveCount()` call ?
I mean avoid this additional `incrementAndGet` on each `beforeHandle`